internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -0,0 +1,532 @@
+			return reconcile.Result{}, nil
+		}
+		r.log.WithError(err).Error("Failed to get resource", req.NamespacedName)
+		return ctrl.Result{}, err
+	}
+
+	if instance.GetDeletionTimestamp().IsZero() {
+		r.log.Infof("Zero deletion timestamp")
+		// Check to see if the ASC has our finalizer, if so, we must add it
+		if !controllerutil.ContainsFinalizer(instance, agentServiceConfigLocalClusterImportFinalizerName) {
+			controllerutil.AddFinalizer(instance, agentServiceConfigLocalClusterImportFinalizerName)
+			if err := r.client.Update(ctx, instance); err != nil {
+				r.log.WithError(err).Error("failed to add finalizer to AgentServiceConfig")
+			}
+		}
+	} else {
+		r.log.Infof("Non zero deletion timestamp")
+		err := r.deleteLocalClusterImportCRs(ctx)
+		if err != nil {
+			r.log.WithError(err).Error("failed to clean up local cluster CRs")
+			return ctrl.Result{Requeue: true}, err
+		}
+		controllerutil.RemoveFinalizer(instance, agentServiceConfigLocalClusterImportFinalizerName)
+		if err := r.client.Update(ctx, instance); err != nil {
+			r.log.WithError(err).Error("failed to remove finalizer from AgentServiceConfig")
+			return ctrl.Result{Requeue: true}, err
+		}
+		// Stop reconciliation as the item is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	hasManagedCluster, err := r.hasLocalManagedCluster(ctx)
+	if err != nil {
+		err = r.setReconciliationStatus(ctx, instance, false, aiv1beta1.ReasonUnableToDetermineLocalClusterManagedStatus, err.Error())
+		if err != nil {
+			r.log.Error("Unable to set reconciliation status of LocalClusterImport on AgentServiceConfig")
+		}
+		return ctrl.Result{Requeue: true}, errors.Wrap(err, "error while attempting to determine presence of managed cluster")
+	}
+	if hasManagedCluster {
+		if err = r.importLocalCluster(ctx, instance); err != nil {
+			return ctrl.Result{Requeue: true}, errors.Wrap(err, "failed to create managed cluster CRs")
+		}
+		err = r.setReconciliationStatus(ctx, instance, true, aiv1beta1.ReasonLocalClusterEntitiesCreated, "")
+		if err != nil {
+			r.log.Error("Unable to set reconciliation status of LocalClusterImport on AgentServiceConfig")
+		}
+	} else {
+		err = r.deleteLocalClusterImportCRs(ctx)
+		if err != nil {
+			return ctrl.Result{Requeue: true}, errors.Wrap(err, "failed to clean up local cluster CRs")
+		}
+		err = r.setReconciliationStatus(ctx, instance, false, aiv1beta1.ReasonLocalClusterEntitiesRemoved, "")
+		if err != nil {
+			r.log.Error("Unable to set reconciliation status of LocalClusterImport on AgentServiceConfig")
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Reconciliation complete
+	return ctrl.Result{}, nil
+}
+
+func checkManagedClusterName(obj metav1.Object) bool {
+	return obj.GetName() == "local-cluster"
+}
+
+func checkSecretName(obj metav1.Object) bool {
+	return obj.GetNamespace() == "openshift-kube-apiserver" && obj.GetName() == "node-kubeconfigs" || obj.GetNamespace() == "openshift-machine-api" && obj.GetName() == "pull-secret"
```

Comment:
I don't know what that is either? What do you want me to do here?

Reply:
I wanted you to investigate if there was a better place to get it from or justify why you think the one in openshift-machine-api is the best option.

I did a quick look in a cluster I have access to and there seems to be a pull secret in `openshift-config` namespace. Additionally the one in `openshift-machine-api` seems to be owned by the provisioning configuration which will not exist on every cluster.

@tsorya @omertuc can either of you confirm if `openshift-config/pull-secret` is the best place to get the pull secret from on the cluster?

Reply:
Yes `openshift-config`/`pull-secret` would be the correct place to look for the pull-secret. It's true that without `Provisioning` then `openshift-machine-api` doesn't even exist. This is the default on all SNOs, so this is indeed a problem.

Having said that, this PR is about moving the controller from the service to the operator, so changing this as part of this PR would be out-of-scope. I recommend a separate PR/ticket, preferably with backports, to change the namespace to `openshift-config`

Reply:
I don't think we need to go as far as backporting.

We [explicitly disabled](https://github.com/openshift/assisted-service/commit/f45f6195b8e5afec03c88b30241325c2aec5b94a) this in previous versions because it caused issues with ACM uninstall so I'm at least treating this as a new implementation.

Reply:
Fair enough